trezorlib/tests/device_tests/test_msg_signtx_zencash.py

```diff
@@ -0,0 +1,93 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2018 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+from binascii import hexlify, unhexlify
+import pytest
+
+from .common import TrezorTest
+from .conftest import TREZOR_VERSION
+
+from trezorlib import coins
+from trezorlib import messages as proto
+from trezorlib.client import CallException
+from trezorlib.tools import parse_path
+
+TxApiTestnet = coins.tx_api['Zencash']
+
+
+TXHASH_5c580d = unhexlify('5c580d7410c626cc3d0b40c8b61fe1902e2fd3b289aae17b16cbc6db16c86b4c')
+TXHASH_adea9e = unhexlify('adea9eace4586b2fba2a6b1dac03fa1099a6c4f2a7ae841e20d973f064c14d85')
+
+
+class TestMsgSigntx(TrezorTest):
+    def test_two_one(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+
+        # tx: 5c580d7410c626cc3d0b40c8b61fe1902e2fd3b289aae17b16cbc6db16c86b4c
+        # input 1: 25000
+        # tx: adea9eace4586b2fba2a6b1dac03fa1099a6c4f2a7ae841e20d973f064c14d85
+        # input 1: 20000
+
+        inp1 = proto.TxInputType(
+            address_n=parse_path("44'/121'/0'/0/0"),  # 1CK7SJdcb8z9HuvVft3D91HLpLC6KSsGb
+            prev_hash=TXHASH_5c580d,
+            prev_index=0,
+            prev_block_hash_bip115=unhexlify('efdca211d7c63deb490e1d4a0c573cef6cc949cc4322075f73c3190100000000'),
+            prev_block_height_bip115=290311,
+        )
+
+        inp2 = proto.TxInputType(
+            address_n=parse_path("44'/121'/0'/0/0"),  # 15AeAhtNJNKyowK8qPHwgpXkhsokzLtUpG
+            prev_hash=TXHASH_adea9e,
+            prev_index=0,
+            prev_block_hash_bip115=unhexlify('fa59243883f41620bb833f73a01b18fee1915e68e805fe640403a02400000000'),
+            prev_block_height_bip115=290314,
+        )
+
+        out1 = proto.TxOutputType(
+            address='znmc8tf4ngHj6YmBr9QNPQMSwu9YwLV6XM6',
+            amount=40000,
+            script_type=proto.OutputScriptType.PAYTOADDRESS,
+            block_hash_bip115=unhexlify('dfdd8aad03f9104a068aae49102926c4d92b1dc38a02b4a471556d2900000000'),
+            block_height_bip115=291331,
+        )
+
+        with self.client:
+            self.client.set_expected_responses([
```

Comment:
Is there any reason to check responses one by one? Does zencash change signature flow from the default btc code -- IOW, would the list of responses be different if you were testing a different coin?

if not, please drop the whole `with` block and just run the `sign_tx` function

Reply:
Yes , you're right, it's no reason to do this validation, I will delete it.